src/nxdoc/web-ui/nuxeo-elements/nuxeo-elements-tutorial.md

```diff
@@ -147,61 +149,93 @@ Let's plug this application into the Nuxeo instance and change the hardcoded use
 
-3.  Add our element to fragments under `polymer.json` so that our element can be loaded lazily:
-
-    {{#> panel type='code' heading='polymer.json'}}
-    ```json
-    {
-      ...
-      "fragments": [
-        ...
-        "src/my-doc-reader.html",
-        ...
-      ],
-      ...
-    }
+3.  Change your routing methods to navigate to the new element:
+
+    {{#> panel type='code' heading='src/my-app.js'}}
+    ```js
+      _routePageChanged(page) {
+        // Show the corresponding page according to the route.
+        //
+        // If no page was found in the route data, page will be an empty string.
+        // Show 'view1' in that case. And if the page doesn't exist, show 'view404'.
+        if (!page) {
+          this.page = 'view1';
+        } else if (['view1', 'view2', 'view3', 'my-doc-reader'].indexOf(page) !== -1) {
+          this.page = page;
+        } else {
+          this.page = 'view404';
+        }
+
+        // Close a non-persistent drawer when the page & route are changed.
+        if (!this.$.drawer.persistent) {
+          this.$.drawer.close();
+        }
+      }
     ```
     {{/panel}}
 
-    {{#> callout type='note' heading='App structure and lazy loading'}}
-    For more information about app structure and lazy loading, please check the
-    [Polymer Documentation](https://polymer-library.polymer-project.org/2.0/docs/apps/prpl).
-    {{/callout}}
+    And add the new route to the `_pageChanged` method:
+
+    {{#> panel type='code' heading='src/my-app.js'}}
+    ```js
+      _pageChanged(page) {
+        // Import the page component on demand.
+        //
+        // Note: `polymer build` doesn't like string concatenation in the import
+        // statement, so break it up.
+        switch (page) {
+          case 'view1':
+            import('./my-view1.js');
+            break;
+          case 'view2':
+            import('./my-view2.js');
+            break;
+          case 'view3':
+            import('./my-view3.js');
+            break;
+          case 'my-doc-reader':
+            import('./my-doc-reader.js');
+            break;
+          case 'view404':
+            import('./my-view404.js');
+            break;
+        }
+      }
+    ```
+    {{/panel}}
 
 ## Plug in Nuxeo Elements
 
 1.  Install Nuxeo elements through Bower:
 
     ```bash
-    # Install bower globally if it is not already done
-    $ npm install -g bower
-    # Add nuxeo-elements as a bower dependency
-    $ bower install --save nuxeo/nuxeo-elements
+    # Add nuxeo-elements as a npm dependency
+    $ npm i @nuxeo/nuxeo-elements
     ```
 
-    This adds `nuxeo-elements` as a dependency in `bower.json` and downloads the latest release from our GitHub [repository](https://github.com/nuxeo/nuxeo-elements) into `bower_components`.
+    This adds `nuxeo-elements` as a dependency in `package.json` and downloads the latest release from our GitHub [repository](https://github.com/nuxeo/nuxeo-elements) into `node_modules`.
```

Comment:
🤔 Doesn't this download the latest release from the npm registry?

Reply:
We should point it to the maintenance 3.x branch, right?